asn1/CommonLedgerProtocol.asn

```diff
@@ -21,81 +21,53 @@ SideProtocolData ::= SEQUENCE OF SEQUENCE {
   protocolData OCTET STRING
 }
 
-PacketRequest ::= SEQUENCE {
-  packet InterledgerPacket,
+Ack ::= SEQUENCE {
+  ack NULL,
```

Comment:
If you look at the way the messages are defined, the logical operations are all defined as `Call`s. Then there is a wrapper that adds the request/response semantics.

The protocol that uses these messages should define "when you send a `Prepare`, you should expect an `Ack` and then later a `Fulfill` or `Reject` as a request".

If you get a request with a non-zero `requestId`, you must send a response or the other party will likely keep sending you the message. If the `requestId` is zero, they'll ignore your response if you send one.

> Are we expecting a server to not know if it's operating in an async or synchronous mode?

That is dependent on the protocol using these messages. In general I would expect all Message requests to be synchronous and all transfers to be asynchronous.

> Are we expecting the receiver to not understand the logical request and so not know if it must ack or respond?

They are expected to understand it. There's an argument to be made that we don't need this extra way to specify whether you need to send a response, but I think it's worth having in there.

> Are we expecting a client to throw an error if it gets an Ack and doesn't want it?

No. @sharafian already observed behavior where triggering an error when you get a message you don't expect can cause an infinite loop if the other side doesn't expect your error message.

> How do I track down a request from the logs if all notifications have the same id (zero)?

If it's a transfer notification you'd look for it by the `transferId`. For quotes you would set a `requestId` because you do want a response.